### PR TITLE
DJS version update and the described changes

### PR DESCRIPTION
**Pull request by RGHANILOO**
following changes applied in the PR
- version update from 13->14
- separated the ready function `ready.ts`
- built a interface for slash command structure `Commands.ts`
- interaction listener implemented `interactionCreator.ts`
- command for hello funnily name `Howdy.ts`

bot is tested on my personal token responds to hello

### CODE REVIEW

File: src/listeners/interactionCreator.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { CommandInteraction, Client, Interaction } from "discord.js";
+import { Commands } from "../commandCenter/Commands";
+
+export default (client: Client): void => {
+    client.on("interactionCreate", async (interaction: Interaction) => {
+        if (interaction.isCommand() || interaction.isContextMenuCommand()) {
+            await handleSlashCommand(client, interaction)
```
Comment (sulphite) [inline src/listeners/interactionCreator.ts]: `interaction.isContextMenuCommand()` shows me a type error on my end - I don't think context menu commands can be handled the same way as slash commands (but haven't spent much time reading, correct me if I'm wrong)? So typescript sees that inside the `if` you're using a function that takes a  `CommandInteraction` but you're also allowing ContextMenuCommand through

Comment (RGHANILOO) [reply src/listeners/interactionCreator.ts]: handleSlash command takes client object type of Client and interaction object type  of CommandInteaction

inside the handler we are checking if the interaction `isCommand()` method or `isContextMenueCommand()`
if it then calls the handleSlashCommand()

[class CommandInteraction](https://discord.js.org/docs/packages/discord.js/main/CommandInteraction:Class#isCommand)

Comment (RGHANILOO) [reply src/listeners/interactionCreator.ts]: reason i had it in there as my understanding was with isCotextCommand. allows us to tap on amessage to use its context in the slash comment
[contextmenu](https://discordjs.guide/interactions/context-menus.html#registering-context-menu-commands)
we don't necessarily need it at this moment though

**Review by sulphite — APPROVED** [review]
sick

**Pull request merged**